core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -252,4 +264,40 @@ private int termLen(String text) {
     public String asString() {
         return fuzziness.toString();
     }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        Fuzziness other = (Fuzziness) obj;
+        return Objects.equals(fuzziness, other.fuzziness);
+    }
+
+    @Override
+    public int hashCode() {
+        return fuzziness.hashCode();
+    }
+    
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeGenericValue(fuzziness);
+    }
+
+    @Override
+    public Fuzziness readFrom(StreamInput in) throws IOException {
+        Object value = in.readGenericValue();
+        if (value instanceof Integer) {
+            return new Fuzziness((Integer) value);
+        } else if (value instanceof Float) {
+            return new Fuzziness((Float) value);
+        } else if (value instanceof String) {
+            return new Fuzziness((String) value);
```

Comment:
Not sure why it is rendered like this in github, but I am using a factory Fuzziness.newFuziness.